### PR TITLE
FIX: pre-composite animation frames to white background

### LINKED ISSUE #19040: v3.3.0 Regression, Animation draws artists multiple times.
### Bug report

**Bug summary**

The Animation classes `ArtistAnimation` and `FunAnimation` over draw onto the figure causing thick lines and "blocky" texts. This is a regression that ships in v3.3.0 (i.e v3.2.2 is good) and is still present in v3.3.3.

**Code for reproduction**
```python
import itertools

import numpy as np
import matplotlib.pyplot as plt
import matplotlib.animation as animation

# for animation in the notebook
from matplotlib import rc
rc('animation', html='html5')

def data_gen():
    for cnt in itertools.count():
        t = cnt / 10
        yield t, np.sin(2*np.pi*t) * np.exp(-t/10.)


def init():
    ax.set_ylim(-1.1, 1.1)
    ax.set_xlim(0, 10)
    del xdata[:]
    del ydata[:]
    line.set_data(xdata, ydata)
    return line,

fig, ax = plt.subplots()
fig.set_size_inches(8, 6)
line, = ax.plot([], [], lw=2)
ax.grid()
xdata, ydata = [], []


def run(data):
    # update the data
    t, y = data
    xdata.append(t)
    ydata.append(y)
    xmin, xmax = ax.get_xlim()

    if t >= xmax:
        ax.set_xlim(xmin, 2*xmax)
        ax.figure.canvas.draw()
    line.set_data(xdata, ydata)

    return line,

ani = animation.FuncAnimation(fig, run, data_gen, interval=10, init_func=init)
ani
```

**Actual outcome (v3.3.3)**

bad-v3 3 3

**Expected outcome (v3.2.2)**

good-v3 2 2

The regression is clearly visible for the ticks and tick labels and less visible (in above example) is that it affects the plotting and legend too.

### PR DESCRIPTION
**Pull request by tacaswell**
This fixes issues with encoding that do not support transparency and when the
Figure facecolor is transparent.

Closes #19040

## PR Summary

I'm not sure how to test this as it is a visual thing is h254 encoded output. 

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [n/a] New features are documented, with examples if plot related.
- [n/a] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by rothn** [comment]
Keep up the great work!

**Review by timhoffm — APPROVED** [review]

File: lib/matplotlib/tests/test_animation.py [hunk 1]
```diff
@@ -82,7 +82,10 @@ def test_null_movie_writer(anim):
     assert writer.outfile == filename
     assert writer.dpi == dpi
     assert writer.args == ()
-    assert writer.savefig_kwargs == savefig_kwargs
+    # we enrich the savefig kwargs to ensure we composite transparent
+    # output to as opaque background
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_animation.py:86]:
```suggestion
    # output to an opaque background
```

File: lib/matplotlib/animation.py [hunk 2]
```diff
@@ -1057,6 +1060,18 @@ def func(current_frame: int, total_frames: int) -> Any
             _log.info("Disabling savefig.bbox = 'tight', as it may cause "
                       "frame size to vary, which is inappropriate for "
                       "animation.")
+
+        facecolor = savefig_kwargs.get('facecolor',
+                                       mpl.rcParams['savefig.facecolor'])
+        if facecolor == 'auto':
+            facecolor = self._fig.get_facecolor()
```
Comment (QuLogic) [inline lib/matplotlib/animation.py:1067]: We should be able to test this code path in the same `NullMovieWriter` test below, no?

Comment (tacaswell) [reply lib/matplotlib/animation.py:1067]: Done, but this is still just a smoke test 😞

**Pull request merged**